packages/node/src/node/client/ClientStructure.ts

```diff
@@ -263,16 +273,47 @@ export class ClientStructure {
 
             // Update version but only if this was a wildcard read
             if (scope.isWildcard(endpointId, clusterId)) {
-                currentUpdates.values[DatasourceCache.VERSION_KEY] = change.version;
+                currentUpdates.values![DatasourceCache.VERSION_KEY] = change.version;
```

Comment:
The "!" assertion is pretty far from where this is set. Maybe test above could be a type assertion so reasoning is more apparent and we don't accidentally break it?  Or just add an if to confirm it's really there